modules/common/secret/secret.go

```diff
@@ -385,3 +385,42 @@ func GetDataFromSecret(
 
 	return data, ctrl.Result{}, nil
 }
+
+// VerifySecret - verifies if the Secret object exists and the expected fields
+// are in the Secret. It returns a hash of the values of the expected fields.
+func VerifySecret(
+	ctx context.Context,
+	secretName types.NamespacedName,
+	expectedFields []string,
+	reader client.Reader,
+	requeueTimeout time.Duration,
+) (string, ctrl.Result, error) {
+	secret := &corev1.Secret{}
+	err := reader.Get(ctx, secretName, secret)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return "",
+				ctrl.Result{RequeueAfter: requeueTimeout},
+				fmt.Errorf("OpenStack secret %s not found", secretName)
```

Comment:
Sorry for noticing just now, but can we please remove OpenStack from it. This is here just because in the nova version we used this method first to check for the osp-secret. But this can be any secret now.